docs/okeps/okep-50088-evpn.md

```diff
@@ -0,0 +1,439 @@
+ip link set vni101 up
+ip link set br101 up
+
+## Create the MAC-VRF
+ip link add br100 type bridge
+ip link set br100 master blue
+# ip link set br100 addr aa:bb:cc:00:00:6e  <-- SVI configuration for the bridge
+# ip addr add 10.0.10.254/24 dev br100      <-- SVI configuration for the bridge
+ip link add vni100 type vxlan local 100.64.0.1 dstport 4789 id 100 nolearning
+ip link set vni100 master br100 addrgenmode none
+ip link set vni100 type bridge_slave neigh_suppress on learning off
+ip link set vni100 up
+ip link set br100 up
+
+## Connect OVS to the linux bridge
+ip link add blue-0 type veth peer name blue-1
+ovs-vsctl add-port br-int blue-0
+ip link set dev blue-1 master br100
+ip link set dev br100 up
+ip link set dev blue-1 up
+```
+
+The MAC addresses for the bridge are unique and will be automatically generated by OVN-Kubernetes. Furthermore, bridge and
+VXLAN link names may also change and will be decided by OVN-Kubernetes. While the IP-VRF uses pure routing to transmit
+traffic over the EVPN fabric, MAC-VRF relies on layer 2. For that reason, the layer 2 OVN network needs to be extended into
+the EVPN fabric. To do that, we connect br-int to the linux bridge for the MAC-VRF. This will allow layer 2 traffic to
+be travel through br100 and then eventually into the EVPN fabric via the VNID 100. This enables a user to disable the
+Geneve overlay and allow L2 communication between UDNs on different nodes via the MAC-VRF:
+
+<a id="l2evpn-anchor"></a>
+![](images/l2evpn.png)
+
+Note, the Layer 2 domain for the MAC-VRF may be extended into the provider's physical network, and not just extended across
+Kubernetes nodes. This allows for VM migration and other layer 2 connectivity between entities external to the cluster
+and entities within.
+
+The SVI interface (br100, br101) requires an IP address only when a MAC-VRF is being used as part of an IP-VRF. This IP
+is used as the next hop for hosts on the layer 2 MAC-VRF to route via the IP-VRF. Since asymmetrical IRB is not supported,
+there is no reason to advertise the SVI IP. Therefore, the SVI IP may be the same value across all Kubernetes nodes.
+Rather than using an SVI interface on the linux bridge, OVN-Kubernetes will leverage the management port to serve as an
+SVI port. The management port is already assigned an IP address that is common across all nodes in a layer 2 network. We
+can route traffic to the ovn_cluster_router and then to the ovn-k8s-mpx management port the same way we do in local gw mode
+today. The packet will then go into the blue VRF and routes learned by BGP will send it via the vni101 VXLAN tunnel.
+
+OVNKube-Controller will be responsible for configuring OVN, including the extra veth port attached to the worker logical
+switch.
+
+In addition to VTEP IP allocation, ovnkube-cluster-manager will be responsible for generating FRR-K8S config to enable
+FRR with EVPN. The config for the above example would look something like this:
+
+```aiignore
+vrf blue
+ vni 101
```

Comment:
does FRRConfiguration CR have support for VNI ? or does it have to enhanced for EVPN ?

Reply:
added a section about FRR-K8S API extension. Will need @jcaamano help here to fill it out.